Source/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs

```diff
@@ -126,6 +126,11 @@ public IFakeOptionsBuilder<T> Implements(Type interfaceType)
                 return this;
             }
 
+            public IFakeOptionsBuilder<T> Implements<TInterface>()
+            {
+                return Implements(typeof (TInterface));

```

Comment:
@thomaslevesque the StyleCop.MSBuild package injects build targets into the
csproj which make StyleCop run every single time you build. You don't need
StyleCop installed on your machine and it doesn't matter if you are in VS
or not.

Anyway,  thats just the nitty gritty of the 'how' not the 'why'. StyleCop
is only there to give us *warnings* when code doesn't meet our guidelines
and a few violations here and there certainly won't stop us accepting a PR.
Of course, we'd prefer PR's to be StyleCop compliant but ultimately we as
owners own the burden of keeping the violation count at  or near to zero.
We don't want this to get in the way of contribution.

On 03:09, Sun, 29 Mar 2015 Blair Conrad <notifications@github.com> wrote:

> In Source/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs
> <https://github.com/FakeItEasy/FakeItEasy/pull/471#discussion_r27350377>:
>
> > @@ -126,6 +126,11 @@ public IFakeOptionsBuilder<T> Implements(Type interfaceType)
> >                  return this;
> >              }
> >
> > +            public IFakeOptionsBuilder<T> Implements<TInterface>()
> > +            {
> > +                return Implements(typeof (TInterface));
>
> Hmm. It lives in the Source\packages. I thought the warnings would show
> up anyhow. Huh.
> Honestly, I don't really understand the thing. It was in the FakeItEasy
> when I showed up, and I've never had any experience with it elsewhere.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FakeItEasy/FakeItEasy/pull/471/files#r27350377>.
>